format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/DelimitedInputFormatProvider.java

```diff
@@ -113,16 +113,21 @@ static Schema detectSchema(DelimitedConfig conf, String delimiter,
       String line;
       String[] columnNames = null;
       String[] rowValue;
+      String pathField = conf.getPathField();
+
       try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputFile.open()))) {
         for (int rowIndex = 0; rowIndex < conf.getSampleSize() && (line = reader.readLine()) != null; rowIndex++) {
+
           rowValue = line.split(delimiter, -1);
           if (rowIndex == 0) {
+
             columnNames = DataTypeDetectorUtils.setColumnNames(line, conf.getSkipHeader(), conf.getEnableQuotedValues(),
-                                                               delimiter);
+                                                               delimiter, pathField);
```

Comment:
the path field is not related to detecting schema from data and should not be passed into the detector. Instead, it should be added to the list of fields that the detector returns.